Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -39,16 +42,16 @@ open class BasicFormat: SyntaxRewriter {
 
   public init(indentationWidth: Trivia = .spaces(4), initialIndentation: Trivia = []) {
     self.indentationWidth = indentationWidth
-    self.indentationStack = [initialIndentation]
+    self.indentationStack = [(indentation: initialIndentation, isUserDefined: false)]
   }
 
   // MARK: - Updating indentation level
 
-  public func pushIndentationLevel(increasingIndentationBy: Trivia) {
-    indentationStack.append(currentIndentationLevel + increasingIndentationBy)
+  public func increaseIndentationLevel() {
+    indentationStack.append((currentIndentationLevel + indentationWidth, false))
   }
 
-  public func popIndentationLevel() {
+  public func decreaseIndentationLevel() {
     precondition(indentationStack.count > 1, "Popping more indentation levels than have been pushed")
```

Comment:
I know this should always be the case since we push on init, but given this is going to be running all the time I would personally prefer this just only remove if count > 1. Then this can be `assert` instead.

Same with the `!` down below in `visitPre` - fallback to increasing indentation level normally if it's `nil` (and assert).